core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -717,6 +719,15 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
 
   def sufficientResourcesRegistered(): Boolean = true
 
+  // When the executor failure tracker collects enough failures, if the current resources are
+  // insufficient for keep the app running, it will fail the application directly; otherwise,
+  // it survives this check round.
+  def insufficientResourcesRetained(): Boolean = {
+    totalRegisteredExecutors.get() < maxExecutors * minSurviveRatio
```

Comment:
with dynamic allocation maxExecutors is Int.MaxValue, so how does that really work with it?  I would basically say it doesn't.

Reply:
When maxExecutors is Int.MaxValue, the value of max failures is 2 * Int.MaxValue or explicit value is given via spark.executor.maxFailures. I guess we don't change the behavior here

Reply:
Exactly so this feature doesn't apply to how many users normally run and could be confusing.

Reply:
I guess Int.MaxValue for maxExecutors is rare. It's problematic itself to use in prod.